crates/provider/src/native.rs

```diff
@@ -0,0 +1,307 @@
+use std::{
+    self,
+    error::Error,
+    fmt::Debug,
+    process::{Command, Output},
+};
+
+use async_trait::async_trait;
+use serde::Serialize;
+
+use crate::shared::{
+    constants::{DEFAULT_DATA_DIR, DEFAULT_REMOTE_DIR, LOCALHOST},
+    provider::Provider,
+    types::{NamespaceDef, NamespaceMetadata, RunCommandOptions, RunCommandResponse},
+};
+
+pub trait FileSystem {
+    fn create_dir(&mut self, path: impl Into<String>) -> Result<(), Box<dyn Error>>;
+    fn write(
+        &mut self,
+        path: impl Into<String>,
+        content: impl Into<String>,
+    ) -> Result<(), Box<dyn Error>>;
+}
+
+#[derive(Debug, Serialize)]
+struct FilesystemInMemory {}
+
+#[derive(Debug, Serialize, Clone, PartialEq)]
+struct NativeProvider<T: FileSystem + Debug> {
+    /// Namespace of the client
+    namespace:             String,
+    /// path where configuration relies
+    config_path:           String,
+    // variable that shows if debug is activated
+    debug:                 bool,
+    /// the timeout for the client to exit
+    timeout:               u32,
+    /// command sent to client
+    command:               String,
+    /// temporary directory
+    tmp_dir:               String,
+    pod_monitor_available: bool,
```

Comment:
Naming convention in Rust is is_something

```suggestion
    is_pod_monitor_available: bool,
```

Reply:
For the reference: https://doc.rust-lang.org/1.0.0/style/style/naming/README.html